Slack doc revamp (#525)
rudderlabs/rudderstack-docs: destinations/business-messaging/slack.mdx

# Slack

> [Slack](https://slack.com/intl/en-in/) is a popular business communication platform that allows you to organize all your business-related chats by specific topics, groups or direct/personal messaging.

Reviewer: ```suggestion
[Slack](https://slack.com/intl/en-in/) is a popular business communication platform that lets you organize all your business-related chats by specific topics, groups, or direct messaging.
```

RudderStack supports Slack as a destination where you can send your event data seamlessly.

<div class="successBlock">

>  Find the open-source transformer code for this destination in our <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/v0/destinations/slack">GitHub repo</a>.

Reviewer: ```suggestion
Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/v0/destinations/slack">GitHub repository</a>.
```

<div class="infoBlock">

To know more about the difference between cloud mode and device mode in RudderStack, refer to the <a href="https://rudderstack.com/docs/rudderstack-cloud/rudderstack-connection-modes/">RudderStack Connection Modes</a> guide.
</div>

Once you have confirmed that the platform supports sending events to Slack, perform the steps below:

1. From your [RudderStack dashboard](https://app.rudderlabs.com/), add the source. Then, from the list of destinations, select **Slack**.
2. Assign a name to the destination and click on **Next**.

### Connection settings

To successfully configure Slack as a destination, you will need to configure the following settings:

<img src="/assets/event-stream-destinations/slack-settings.png" alt="Slack connection settings" />

#### Event Channels

- **Event Name**: Enter the event name or regex expression to match the RudderStack event name.

> - **Event Channel**: Slack channel where the event will be sent. You can specify a `#channel_name` or `@user_name`.

Reviewer: ```suggestion
- **Event Channel**: Enter the name of the Slack channel where the event will be sent. You can specify a `#channel_name` or `@user_name`.
```

- **Regex Matching**: Enable this setting if the event name is a regex in the first parameter.                 

<div class="warningBlock">

RudderStack adds the global \(`g`\) parameter implicitly. Hence, you need not add it with the regex expression.
</div>

#### Identify Template

<img src="/assets/event-stream-destinations/slack-settings4.png" alt="Slack connection settings" />

> **Identify Template:** Provide the template that you want the `identify` event to be transformed to when sent to Slack. The default template is:

Reviewer: ```suggestion
**Identify Template**: Specify the template that you want the `identify` event to be transformed to before it is sent to Slack. The default template is shown below:
```

> `Identified {{name}} <traits_key1>:<traits_value1> <traits_key2>:<traits_value2> ....`

Reviewer: ```suggestion
```
Identified {{name}} <traits_key1>:<traits_value1> <traits_key2>:<traits_value2> ....`
```
```

> `Identified {{name}} <traits_key1>:<traits_value1> <traits_key2>:<traits_value2> ....`

Reviewer: We can specify this template in a code block rather than presenting it as code in text. WDYT? It will be easier to copy that way.

 
   `{{name}} did {{event}}` 

   Here, the `name` is determined in the same way as above.
- **Regex Matching:**  Enable this setting if the event name is a regex in the first parameter.

#### Webhook URL

<img src="/assets/event-stream-destinations/slack-settings3.png" alt="Slack connection settings" />

**Webhook URL:** Mention the Slack's incoming [Webhook URL](https://my.slack.com/services/new/incoming-webhook/). To obtain the Webhook URL, follow the below steps:

1. Click on your Slack workspace name, then go to **Administration** > **Manage Apps**.
2. Search for **Incoming webhooks** in the Slack app directory and add it to Slack by clicking on **Add to Slack**.
3. Select a channel where you want to post the messages forwarded by webhook and click on **Add Incoming webhooks integration**.

<img src="/assets/event-stream-destinations/choose-channel-slack.png" alt="Slack connection settings" />

4. Copy the webhook URL and enter it in the RudderStack dashboard.

<img src="/assets/event-stream-destinations/webhook-url-slack.png" alt="Slack connection settings" />

## Handlebar Expression

> RudderStack supports [Handlebar expression](https://handlebarsjs.com/guide/expressions.html) by leveraging the Handlebar.js library for including the variables into your dynamic Slack messages.

Reviewer 2: It should be handlebars.js library.

 
```
{{name}} did {{event}}
```

Here, `name` is determined in the same way as described in the [Identify Template](#identify-template) section above.

- **Regex Matching**:  Enable this setting if the event name is a regex in the first parameter.

#### Webhook URL

<img src="../../assets/event-stream-destinations/slack-settings3.png" alt="Slack connection settings" />

**Webhook URL:** Enter your Slack's [incoming webhook URL](https://my.slack.com/services/new/incoming-webhook/). 

To obtain the webhook URL, follow the below steps:

1. Click on your Slack workspace name, then go to **Administration** > **Manage Apps**.
2. Search for **Incoming webhooks** in the Slack app directory and add it to Slack by clicking on **Add to Slack**.
3. Select a channel where you want to post the messages forwarded by the webhook and click on **Add Incoming webhooks integration**.

<img src="../../assets/event-stream-destinations/choose-channel-slack.png" alt="Slack connection settings" />

4. Finally, copy the webhook URL and enter it in the RudderStack dashboard.

<img src="../../assets/event-stream-destinations/webhook-url-slack.png" alt="Slack connection settings" />

> ## Handlebar expressions

Reviewer: @utsabc @ranganathkoman please confirm if these handlebar expressions are helpful when defining the identify and event templates described in the **Connection Settings** section above? We can add a cross-reference then for better clarity. cc @ishachopra0401
